### PR TITLE
[FLINK-34686][cdc] add docs about yarn deployment

### PR DESCRIPTION
**Pull request by Shawn-Hx**
Add doc about deploying CDC pipeline to yarn cluster.

### CODE REVIEW

File: docs/content/docs/deployment/yarn.md [hunk 1]
```diff
@@ -23,3 +23,131 @@ KIND, either express or implied.  See the License for the
...
+echo "stop" | ./bin/yarn-session.sh -id application_XXXXX_XXX
+```
+
+After starting YARN session, you can now access the Flink Web UI through the URL printed in the last lines of the command output, or through the YARN ResourceManager web UI.
+
+Then, you need to add two configs to your flink-conf.yaml:
+
+```yaml
+rest.bind-port: {{REST_PORT}}
+rest.address: {{NODE_IP}}
```
Comment (PatrickRen) [inline docs/content/docs/deployment/yarn.md:87]:
```suggestion
rest.bind-port: {{REST_PORT}}
rest.address: {{NODE_IP}}
execution.target: yarn-session
yarn.application.id: {{YARN_APPLICATION_ID}}
```

File: docs/content/docs/deployment/yarn.md [hunk 2]
```diff
@@ -23,3 +23,131 @@ KIND, either express or implied.  See the License for the
...
+```
+
+{{REST_PORT}} and {{NODE_IP}} should be replaced by the actual values of your JobManager Web Interface.
```
Comment (PatrickRen) [inline docs/content/docs/deployment/yarn.md:90]:
```suggestion
{{REST_PORT}} and {{NODE_IP}} should be replaced by the actual values of your JobManager Web Interface, and {{YARN_APPLICATION_ID}} should be replaced by the actual YARN application ID of Flink.
```

File: docs/content/docs/deployment/yarn.md [hunk 3]
```diff
@@ -23,3 +23,131 @@ KIND, either express or implied.  See the License for the
...
+
+### Set up Flink CDC
+Download the tar file of Flink CDC from [release page](https://github.com/apache/flink-cdc/releases), then extract the archive:
+
+```bash
+tar -xzf flink-cdc-*.tar.gz
+```
+
+flink-cdc directory will contain four directories: `bin`,`lib`,`log`,`conf`.
```
Comment (PatrickRen) [inline docs/content/docs/deployment/yarn.md:99]:
```suggestion
Extracted `flink-cdc` contains four directories: `bin`,`lib`,`log` and `conf`.
```

File: docs/content/docs/deployment/yarn.md [hunk 4]
```diff
@@ -23,3 +23,131 @@ KIND, either express or implied.  See the License for the
...
+
+Download the connector package listed below and move it to the `lib` directory.    
+**Download links are available only for stable releases, SNAPSHOT dependencies need to be built based on master or release branches by yourself.**
+- [MySQL pipeline connector 3.0.0](https://repo1.maven.org/maven2/org/apache/flink/flink-cdc-pipeline-connector-mysql/3.0.0/flink-cdc-pipeline-connector-mysql-3.0.0.jar)
+- [Apache Doris pipeline connector 3.0.0](https://repo1.maven.org/maven2/org/apache/flink/flink-cdc-pipeline-connector-doris/3.0.0/flink-cdc-pipeline-connector-doris-3.0.0.jar)
```
Comment (PatrickRen) [inline docs/content/docs/deployment/yarn.md:104]: What about pointing to the `Connectors` page? It's hard to list all connectors here

File: docs/content/docs/deployment/yarn.md [hunk 5]
```diff
@@ -23,3 +23,131 @@ KIND, either express or implied.  See the License for the
...
+
+After successful submission, the return information is as follows：
+
+```bash
+Pipeline has been submitted to cluster.
+Job ID: ae30f4580f1918bebf16752d4963dc54
+Job Description: Sync MySQL Database to Doris
+```
+
+We can find a job  named `Sync MySQL Database to Doris` is running through Flink Web UI.
```
Comment (PatrickRen) [inline docs/content/docs/deployment/yarn.md:151]:
```suggestion
You can find a job named `Sync MySQL Database to Doris` running through Flink Web UI.
```

**Review by PatrickRen — COMMENTED** [review]
@Shawn-Hx Thanks for the PR! I left some comments.

File: docs/content/docs/deployment/yarn.md [hunk 6]
```diff
@@ -23,3 +23,131 @@ KIND, either express or implied.  See the License for the
...
+ parallelism: 2
+
+```
+
+You need to modify the configuration file according to your needs.
+Finally, submit job to Flink Standalone cluster using Cli.
+
+```bash
+cd /path/flink-cdc-*
+./bin/flink-cdc.sh mysql-to-doris.yaml
```
[the remark this answers is not in the document]
Comment (leonardBang) [reply docs/content/docs/deployment/yarn.md:140]: We simply use`$> flink-cdc.sh mysql-to-doris.yaml`

[the remark this answers is not in the document]
Comment (Shawn-Hx) [reply docs/content/docs/deployment/yarn.md:140]: The file `flink-cdc.sh` is in the bin directory, you mean we should use `cd /path/flink-cdc-*/bin` then submit yaml ?

[the remark this answers is not in the document]
Comment (leonardBang) [reply docs/content/docs/deployment/yarn.md:140]: I got your point after read https://nightlies.apache.org/flink/flink-docs-master/docs/try-flink/local_installation/#starting-and-stopping-a-local-cluster

**Review by leonardBang — APPROVED** [review]
+1

**Review by PatrickRen — APPROVED** [review]
@Shawn-Hx Thanks for the update. LGTM

**Pull request merged**